pkg/cmd/release-payload-controller/payload_creation_controller.go

```diff
@@ -0,0 +1,172 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"reflect"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// These are valid Reason values for the ReleasePayloadStatus Conditions
+const (
+	// ReleasePayloadCreatedReason programmatic identifier indicating that the ReleasePayload created successfully
+	ReleasePayloadCreatedReason string = "ReleasePayloadCreated"
+
+	// ReleasePayloadFailedReason programmatic identifier indicating that the ReleasePayload creation failed
+	ReleasePayloadFailedReason string = "ReleasePayloadFailed"
+)
+
+// PayloadCreationController is responsible for setting the PayloadCreated and PayloadFailed conditions based
+// on whether the release payload creation job completed successfully or not.
+// The PayloadCreationController reads the following pieces of information:
+//   - .status.releaseCreationJobResult.status
+// and populates the following conditions:
+//   - .status.conditions.PayloadCreated
+//   - .status.conditions.PayloadFailed
+type PayloadCreationController struct {
+	*ReleasePayloadController
+}
+
+func NewPayloadCreationController(
+	releasePayloadNamespace string,
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	eventRecorder events.Recorder,
+) (*PayloadCreationController, error) {
+	c := &PayloadCreationController{
+		ReleasePayloadController: NewReleasePayloadController("Payload Creation Controller",
+			releasePayloadNamespace,
+			releasePayloadInformer,
+			releasePayloadClient,
+			eventRecorder.WithComponentSuffix("payload-creation-controller"),
+			workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "PayloadCreationController"),
+			false),
+	}
+
+	c.syncFn = c.sync
+
+	releasePayloadFilter := func(obj interface{}) bool {
+		if releasePayload, ok := obj.(*v1alpha1.ReleasePayload); ok {
+			// If the conditions are both in their respective terminal states, then there is nothing else to do...
+			if (conditions.IsConditionTrue(releasePayload.Status.Conditions, v1alpha1.ConditionPayloadCreated) && conditions.IsConditionFalse(releasePayload.Status.Conditions, v1alpha1.ConditionPayloadFailed)) ||
+				(conditions.IsConditionFalse(releasePayload.Status.Conditions, v1alpha1.ConditionPayloadCreated) && conditions.IsConditionTrue(releasePayload.Status.Conditions, v1alpha1.ConditionPayloadFailed)) {
+				return false
+			}
+			if len(releasePayload.Status.ReleaseCreationJobResult.Status) > 0 {
+				return true
+			}
+		}
+		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+			if releasePayload, ok := tombstone.Obj.(*v1alpha1.ReleasePayload); ok {
+				// If the conditions are both in their respective terminal states, then there is nothing else to do...
```

Comment:
if it's deleted, you have no work to do, right?

Reply:
Correct.  Removed checks and added log message.
Done